core/src/cubos/input/input_manager.cpp

```diff
@@ -8,6 +8,40 @@ using namespace cubos::input;
 void InputManager::Init(cubos::io::Window* window)
 {
     InputManager::window = window;
+    InputManager::window->onKeyDown.registerCallback(handleKeyDown);
+    InputManager::window->onKeyUp.registerCallback(handleKeyUp);
+    InputManager::window->onMouseDown.registerCallback(handleMouseDown);
+    InputManager::window->onMouseUp.registerCallback(handleMouseUp);
+}
+
+void InputManager::handleKeyDown(Key key)
+{
+    if (InputManager::keyBindings.empty() || !InputManager::keyBindings.contains(key) ||
+        InputManager::keyBindings[key].empty())
+    {
+        return;
+    }
+
+    for (std::vector<std::function<void(void)>>::iterator it = InputManager::keyBindings[key].begin();

```

Comment:
Here you can just use `auto it = ...`, there's no need to specify the whole type.